runtime/src/append_vec.rs

```diff
@@ -433,6 +442,11 @@ impl AppendVec {
                 break;
             }
         }
+
+        // The last entry in this offset needs to be the u64 aligned offset, because that's
+        // where the *next* entry will begin to be stored.
+        rv.push(u64_align!(*offset));
```

Comment:
hehe this is good indication writing tests can expose bugs. :)

Reply:
Hehe yeah, this was a good one!